src/TTextEdit.cpp

```diff
@@ -1674,11 +1674,31 @@ int TTextEdit::bufferScrollDown( int lines )
 
         return lines;
     }
-    else
+    else if( mpBuffer->mCursorY >= (int)(mpBuffer->size()-1) )
     {
         mIsTailMode = true;
         mpBuffer->mCursorY = mpBuffer->lineBuffer.size();
         forceUpdate();
         return 0;
     }
+    else
+    {
+        lines = (int)(mpBuffer->size()-1) - mpBuffer->mCursorY;
```

Comment:
Is it wise to assign a _new_ value to the __supplied argument__ - I know it can be done but is the effect what is wanted?